hw03_frequency_analysis/top.go

```diff
@@ -1,6 +1,55 @@
 package hw03_frequency_analysis //nolint:golint,stylecheck
 
-func Top10(_ string) []string {
-	// Place your code here
-	return nil
+import (
+	"regexp"
+	"sort"
+)
+
+const top = 10
+
+type WordStatistic struct {
+	Word  string
+	Count int
+}
+
+func splitTextToParts(inputText string) []string {
+	return regexp.MustCompile(`[\s\t\r\n]+`).Split(inputText, -1)
+}
+
+func Top10(inputText string) []string {
+	if len(inputText) == 0 {
+		return nil
+	}
+
+	parts := splitTextToParts(inputText)
+
+	wordsList := make(map[string]int)
+	for _, part := range parts {
+		if len(part) > 0 {
+			wordsList[part]++
+		}
+	}
+
+	words := []WordStatistic{}
+	for word, count := range wordsList {
+		words = append(words, WordStatistic{word, count})
+	}
+
+	sort.Slice(words, func(i, j int) bool {
+		return words[i].Count > words[j].Count
+	})
+
+	var limit int
+	if len(words) < top {
+		limit = len(words)
+	} else {
+		limit = top
+	}
+
+	result := []string{}
```

Comment:
```
var result []string
```
https://github.com/uber-go/guide/blob/master/style.md#nil-is-a-valid-slice

Reply:
++